### PR TITLE
Please review: Implemented exercise nucleotide-count

### PR DESCRIPTION
**Pull request by Vankog**
WIP
edit: done(?)

### CODE REVIEW

**Comment by Vankog** [comment]
Well it seems I'm done. Please review!

File: config.json [hunk 1]
```diff
@@ -1,1052 +1,1065 @@
 {
```
Comment (Vankog) [inline config.json]: Ran the configlet fmt as described in the documentation:
https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md#configuring-the-exercise
It seems the configlet wasn't run in quite a time. Or the formatter changed drastically... don't know right now.

File: config.json [hunk 2]
```diff
@@ -1,1052 +1,1065 @@
...
+        "control-flow-(loops)",
+        "mathematics",
+        "randomness",
+        "strings",
+        "text-formatting",
+        "transforming"
+      ],
+      "uuid": "62d60b42-93bc-4de9-90d1-1ca18a847812"
     },
     {
```
Comment (Vankog) [inline config.json]: I put it here, because it is related to the rna-transcriptor. That's why I let it unlock with that exercise.
The uuid I generated with https://www.uuidgenerator.net

File: exercises/nucleotide-count/package.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
+{
```
Comment (Vankog) [inline exercises/nucleotide-count/package.json]: I updated and customized the package.json

File: exercises/nucleotide-count/nucleotide-count.spec.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  xit('counts 0 for unknown or invalid nucleotides.', () => {
+    const strand = 'ACGTACGT';
+
+    // U-nucleotide as part of the RNA
+    expect(new Dna(strand).count('U')).toEqual(0);
+    // no-real-nucleotide
+    expect(new Dna(strand).count('Z')).toEqual(0);
+    // special-character-nucleotides
+    expect(new Dna(strand).count('Ä')).toEqual(0);
+    expect(new Dna(strand).count('×')).toEqual(0);
```
Comment (Vankog) [inline exercises/nucleotide-count/nucleotide-count.spec.js]: inserting an Unicode character by design.

File: config/maintainers.json [hunk 5]
```diff
@@ -1,35 +1,35 @@
 {
```
Comment (Vankog) [inline config/maintainers.json]: formatted by configlet fmt.
Same reasoning as above.

**Comment by tejasbubane** [comment]
Hi @Vankog. Thanks a lot for your contribution! 👍  

Could you please sync up the tests with the canonical tests [here](https://github.com/exercism/problem-specifications/blob/master/exercises/nucleotide-count/canonical-data.json)?
These are the common set of tests we prefer to adhere to with all languages in exercism.

File: config.json (hunk 1 above)
Comment (rchavarria) [reply config.json]: Hmm. The configlet does many changes to this file, and they seem out of the scope of this MR.

We should handle configlet changes in a new PR (it'll perfectly fine if you run it and create that PR).

So, I'd leave this file untouched, just modifications related to `nucleotide-count`. You can do those changes *compatible* with the new configlet. It'll be ok. But, I wouldn't change the rest of exercises configuration in the PR.

What do you think?

File: config/maintainers.json (hunk 5 above)
Comment (rchavarria) [reply config/maintainers.json]: It could be included in the new PR with configlet changes.

File: exercises/nucleotide-count/package.json (hunk 3 above)
Comment (rchavarria) [reply exercises/nucleotide-count/package.json]: `package.json` is kept in sync with the one in the root folder. All exercises have the same `package.json`. If this is different, well lose your customizations. Is there anything special (different from the *main* one) you need?

File: config.json (hunk 1 above)
Comment (Vankog) [reply config.json]: Yeah, actually I did so in the JS track with a separate PR that got merged just recently. Bit this epiphany struck me too late for thes ecma track. ;-)
I'll splitt it.

File: exercises/nucleotide-count/package.json (hunk 3 above)
Comment (Vankog) [reply exercises/nucleotide-count/package.json]: I see, in that case I'll revert it. It just includes some new versions and a customized description and authoring.

File: config/maintainers.json (hunk 5 above)
Comment (Vankog) [reply config/maintainers.json]: dito

**Comment by Vankog** [comment]
Alright, thanks for your feedback. Added some replies to the reviews.
I agree with splitting the configlet fmt into another PR. I'll do so.

About the tests: In the JS track I just got the same advice about the canonical versions.
I'll probably will open a discussion in the canonical track to discuss some changes, because I think some canonical test cases in the different exercises might be bit suboptimal. Firstly, they should provide an optimal B/TDD experience to teach about. It's a learning platform after all. Secondly, some do not seem to cover all suitable edge cases properly, but we have to look closer to the specifics to judge.

**Comment by tejasbubane** [comment]
> About the tests: In the JS track I just got the same advice about the canonical versions.
> I'll probably will open a discussion in the canonical track to discuss some changes, because I think some canonical test cases in the different exercises might be bit suboptimal. Firstly, they should provide an optimal B/TDD experience to teach about. It's a learning platform after all. Secondly, some do not seem to cover all suitable edge cases properly, but we have to look closer to the specifics to judge.

Yes, even I felt the same while looking at the tests for this exercise. Please open a discussion there.

File: config.json [hunk 6]
```diff
@@ -53,6 +53,19 @@
...
+      "topics": [
+        "filtering",
+        "pattern-matching",
+        "strings",
+        "test-driven-development"
+      ],
+      "unlocked_by": "rna-transcription",
+      "uuid": "9debee8e-003d-4eb2-bf97-6d3764d024ca"
+    },
+    {
```
Comment (Vankog) [inline config.json]: I put it here, because it is related to the rna-transcriptor. That's why I let it unlock with that exercise.
The uuid I generated with https://www.uuidgenerator.net

**Comment by Vankog** [comment]
OK, reverted the configlet formatting for this PR and removed the package.json.

I also created a discussion PR for the canonical test:
https://github.com/exercism/problem-specifications/pull/895
I have yet to adjust my test accordingly and I'll wait for what it brings. Till then this PR is on hold.

File: config.json (hunk 1 above)
Comment (rchavarria) [reply config.json]: 👍

**Comment by tejasbubane** [comment]
`package.json` should be added for this exercise. 

I know this PR is on hold, but something I just noticed and commenting so it does not miss out later.

**Comment by matthewmorgan** [comment]
@Vankog what is the status of this PR?  It looks like your PR to problem-specifications has been closed?  Should we close this too, or are you intending to finish it?

Thanks!

**Comment by Vankog** [comment]
no its just closed, because it was splitted.

see https://github.com/exercism/problem-specifications/pull/895#issuecomment-336448399

However, until next year I am not able to work on this. Hot phase for my masters thesis... ;-)

**Comment by matthewmorgan** [comment]
OK, we're going to close this PR for now.  Thanks for the update.
